crates/core/src/subscription/module_subscription_actor.rs

```diff
@@ -185,45 +181,37 @@ impl ModuleSubscriptions {
         &self,
         client: Option<&ClientConnectionSender>,
         mut event: ModuleEvent,
-        ctx: &ExecutionContext,
         tx: MutTx,
     ) -> Result<Result<Arc<ModuleEvent>, WriteConflict>, DBError> {
         // Take a read lock on `subscriptions` before committing tx
         // else it can result in subscriber receiving duplicate updates.
         let subscriptions = self.subscriptions.read();
         let stdb = &self.relational_db;
-
         // Downgrade mutable tx.
         // Ensure tx is released/cleaned up once out of scope.
         let read_tx = scopeguard::guard(
             match &mut event.status {
                 EventStatus::Committed(db_update) => {
-                    let Some((tx_data, read_tx)) = stdb.commit_tx_downgrade(ctx, tx)? else {
+                    let Some((tx_data, read_tx)) = stdb.commit_tx_downgrade(tx, Workload::Subscribe)? else {
```

Comment:
Note: This should be `Workload::Update` for incremental update. `Workload::Subscribe` is for the initial query.